msaad.go

```diff
@@ -870,7 +899,9 @@ func (m *MSAAD) doHTTP(request *http.Request) (*http.Response, error) {
 
 	request.Header.Set("Authorization", "Bearer "+token)
 
-	return http.DefaultClient.Do(request)
+	client := http.DefaultClient
+	client.Timeout = 5 * time.Second
```

Comment:
That is a tight deadline, I am going to assume there is a good reason for this.

Reply:
Purely because there was no deadline in place prior...suggestions?